app/controllers/account_reset/cancel_controller.rb

```diff
@@ -1,9 +1,24 @@
 module AccountReset
   class CancelController < ApplicationController
+    before_action :check_feature_enabled
+
+    def show
+      return render :show unless token
+
+      result = AccountReset::ValidateCancelToken.new(token).call
+      track_event(result)
+
+      if result.success?
+        handle_valid_token
+      else
+        handle_invalid_token(result)
+      end
+    end
+
     def create
-      result = AccountReset::Cancel.new(params[:token]).call
+      result = AccountReset::Cancel.new(session[:cancel_token]).call
 
-      analytics.track_event(Analytics::ACCOUNT_RESET, result.to_h)
+      track_event(result)
 
       handle_success if result.success?
```

Comment:
I know this was here before, but I wonder if we can take this opportunity to make the experience better for the user by displaying the flash error message if they happen to end up on the confirm screen without a token? For example, if somehow there is a bug and the link in the email doesn't include the token, they end up on the `account_reset/cancel` page, and when they click the button to confirm, they get redirected to the sign in page without any kind of message, which could confuse the user.

Reply:
I believe the message they would get is Invalid token on the post.  They do get the confirm screen with no token.  The same thing would occur for truncated link/token.  I believe this is how delete works now too.

Reply:
Agreed. I'm talking about an edge case where someone ended up visiting the `account_reset/cancel` path without any token in the params. In that case, no error would show up because of the first line `render :show and return unless token`. Then, if they submit the form, they will get redirected to the sign in page without an error message. So, instead of this:
```ruby
def create
  result = AccountReset::Cancel.new(session[:cancel_token]).call
  track_event(result)
  handle_success if result.success?
  redirect_to root_url
end
```
we could do something like this:
```ruby
def create
  result = AccountReset::Cancel.new(session[:cancel_token]).call

  track_event(result)
  sign_out if current_user

  if result.success?
    flash[:success] = t('two_factor_authentication.account_reset.successful_cancel')
  else
    flash[:error] = result.errors[:token].first
  end

  redirect_to root_url
end
```

Since this isn't directly related to this PR, I'll leave it up to you if you want to do it now or in another PR.

Reply:
Yes.  I confirmed there is no message on the post after no token.  I thought there was.  I'll update this.